aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -195,6 +195,59 @@ Tensor pinverse(const Tensor& self, double rcond) {
   return at::linalg_pinv(self, rcond, /*hermitian=*/false);
 }
 
+Tensor& linalg_matrix_power_out(const Tensor& self, int64_t n, Tensor& result) {
+  checkSameDevice("matrix_power", result, self);
+  checkLinalgCompatibleDtype("matrix_power", result, self);
+  at::native::resize_output(result, self.sizes());
+  result.copy_(at::native::linalg_matrix_power(self, n));
```

Comment:
If we have to keep this pattern let's use this opportunity to implement a safe copy function. Can we not fuse this copy, however?

Reply:
Implemented out version.